plugin/src/main/groovy/com/btkelly/gnag/utils/ProjectHelper.groovy

```diff
@@ -36,21 +36,60 @@ class ProjectHelper {
         return project.getExtensions().findByName("android") != null;
     }
 
-    public List<File> getSources() {
+    public List<File> getJavaSourceFiles() {
+        return getSourceFilesWithSuffices(Collections.singletonList(".java"))
+    }
+
+    public List<File> getKotlinSourceFiles() {
+        return getSourceFilesWithSuffices(Arrays.asList(".kt", ".kts"))
+    }
+
+    public File getReportsDir() {
+        File reportsDir = new File(project.buildDir.path + "/outputs/gnag/")
+        reportsDir.mkdirs()
+        return reportsDir
+    }
+
+    public File getKtlintReportFile() {
+        return new File(getReportsDir(), "ktlint_report.xml")
+    }
+
+    private Collection<File> getSourceFilesWithSuffices(final List<String> suffices) {
+        final Collection<File> allSourceFiles
+
         if (isAndroidProject()) {
-            project.android.sourceSets.inject([]) {
-                dirs, sourceSet -> dirs + sourceSet.java.srcDirs
+            allSourceFiles = project.android.sourceSets.inject([]) {
+                dirs, sourceSet -> dirs + sourceSet.allSource
             }
         } else {
-            project.sourceSets.inject([]) {
-                dirs, sourceSet -> dirs + sourceSet.java.srcDirs
+            allSourceFiles = project.sourceSets.inject([]) {
+                dirs, sourceSet -> dirs + sourceSet.allSource
+            }
+        }
+
+        return allSourceFiles.findAll { File file ->
+            file.exists() && suffices.any { suffix ->
+                file.name.endsWith(suffix)
             }
         }
     }
 
-    public File getReportsDir() {
-        File reportsDir = new File(project.buildDir.path + "/outputs/gnag/")
-        reportsDir.mkdirs()
-        return reportsDir
+    Set<String> getSourceSetRootDirPaths() {
+        final Set<File> allSourceSetRootDirs
+
+        if (isAndroidProject()) {
+            allSourceSetRootDirs = project.android.sourceSets.inject([]) {
+                dirs, sourceSet -> dirs + sourceSet.allSource.srcDirs
+            }
+        } else {
+            allSourceSetRootDirs = project.sourceSets.inject([]) {
+                dirs, sourceSet -> dirs + sourceSet.allSource.srcDirs
+            }
+        }
+
+        return allSourceSetRootDirs
+                .findAll { File sourceSetRootDir -> sourceSetRootDir.exists() }
```

Comment:
Some default source sets can be defined, but unused (e.g. `resources` in our example).